src/java.desktop/windows/classes/com/sun/java/swing/plaf/windows/WindowsInternalFrameTitlePane.java

```diff
@@ -85,6 +85,7 @@ protected void installDefaults() {
                 buttonWidth = (int) ((float) buttonWidth * d.width / d.height);
             }
         } else {
+            buttonWidth = buttonHeight;
```

Comment:
This means line 71 can be removed because `buttonWidth` is never read from `UIManager`.

Reply:
It can be removed in the strict sense but we need something to initialise it to so kept it as it is, as it will do no harm

Reply:
> It can be removed in the strict sense but we need something to initialise it to so kept it as it is, as it will do no harm

Why does it need anything to initialise? The value is now set in either `if` or `else` branch.

It does not harm, yet it's never used. Not initialising it with `"InternalFrame.titleButtonWidth"` conveys it clearer: the value is computed later.

Reply:
ok. I thought buttonWidth was declared in the method but it is class variable

Reply:
Nothing would've changed if it had been a local variable. Both branches initialise the local variable, the compiler is happy.